src/xsyspath.cpp

```diff
@@ -0,0 +1,38 @@
+/***************************************************************************
+* Copyright (c) 2018, Martin Renou, Johan Mabille, Sylvain Corlay, and     *
+* Wolf Vollprecht                                                          *
+* Copyright (c) 2018, QuantStack                                           *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#include "pybind11/pybind11.h"
+
+#include <iostream>
+#include <string>
+
+#include "xeus-python/xeus_python_config.hpp"
+
+#include "xsyspath.hpp"
+#include "xpaths.hpp"
+
+namespace xpyt
+{
+    void set_syspath()
+    {
+// The XEUS_PYTHON_SYSPATH compile-time definition can be used.
+// to specify the syspath locations as relative paths to the PREFIX.
+#if defined(XEUS_PYTHON_SYSPATH)
+        static const std::string syspath = XPYT_STRINGIFY(XEUS_PYTHON_SYSPATH);
+        // TODO: prefix each item in the ';' or ':' separated list with prefix_path()
+#if PY_MAJOR_VERSION == 2
+        Py_SetPath(const_cast<char*>(syspath.c_str()));
+#else
+        static const std::wstring wstr(syspath.cbegin(), syspath.cend());;
+        Py_SetPath(const_cast<wchar_t*>(wstr.c_str()));
+#endif
+#endif
```

Comment:
As per the `TODO:` above, I presume we would specify the paths in `XEUS_PYTHON_SYSPATH` in a relative fashion with respect to the installation prefix.

Reply:
Maybe the way to go would be to do a replacement of a magic string, like `SYS_PREFIX`.